### DISCUSSION TITLE
Kompatibilität Daikin ECH2O mit RoCon BM2C

### DISCUSSION THREAD

**Discussion by Nitrox912** [Trunks1982/Daikin-Rotex-HPSU-CAN-Seriell#20] [Q&A]
Hallo, tolles Projekt, welches ich gerne umsetzen und in meiner Home Assistant Instanz einsetzen würde. Bevor ich jedoch mit der Teilebestellung starte die Frage: Funktioniert das System analog mit der RoCon BM2C Platine? CAN-H und CAN-L am Anschluss J13 sind zumindest wie bei der BM1 vorhanden.

Viele Grüße

**Comment by Trunks1982** [comment]
Für mich sieht das so aus als würde es funktionieren.Bedienteil sieht auch wie bei den Rotex anlagen aus.

**Comment by Nitrox912** [comment]
Bei dem Fhem Projekt (https://wiki.fhem.de/wiki/Rotex_HPSU_Compact) wird am Anfang unter Allgemein zumindest beide Schnittstellen erwähnt. Demzufolge sollte die Kommunikationsschnittstelle auch gleich sein. Es ist also einen Versuch wert! :)

Ich fahre meine Anlage mit fixer Vorlauftemperatur. Besteht die Möglichkeit diese in Zukunft auch über das System anzupassen?

**Reply by Trunks1982** [reply]
Ich müsste nur diese Punkte einpflegen.Also sollte kein Problem sein.Darf ich fragen wieso mit einer fixen Temp?
Ist ja eigentlich unüblich.

**Reply by Nitrox912** [reply]
Ich habe eine 24-kWp-Photovoltaikanlage auf dem Dach installiert und nutze, soweit möglich (von Februar bis November), die Sonnenenergie ausschließlich tagsüber, um den Eigenverbrauch zu maximieren. Dabei stelle ich die Heizung immer 2-3 Grad höher ein als eigentlich notwendig, um den Estrich als Wärmespeicher zu verwenden. Aufgrund seiner Trägheit funktioniert dies sehr gut, und die Wärme hält bis zum nächsten Morgen an. Sobald die Sonne aufgeht, beginnt die Wärmepumpe erneut zu arbeiten. Zwar verbrauche ich insgesamt mehr Energie durch das ständige Aufheizen und Abkühlen, diese ist jedoch "kostenlos" durch die Photovoltaikanlage gewonnen. Mit der fixen Vorlautemperatur habe ich hierbei einfach mehr Kontrolle über das System und das funktioniert ganz gut.

**Reply by Trunks1982** [reply]
Ah ok verstehe.Da reicht ja in der regel die verstellung von min vl.
Bzw kann man das aber auch über den Raumsoll machen.Sg Schnittstelle macht das ja mit 5k.Müsste gleiche sein als wenn man raumsoll hochstellt.
Ich sag mal da ist man ja ziemlich frei wie man das macht,

**Reply by Nitrox912** [reply]
Über die SG-Schnittstelle sind das aber 5 K WW-Temperatur. Trotzdem eine gute Idee das über den SG-Kontakt zu lösen. Dieser lässt sich aber auch über dein System schalten, sodass ich den Kontakt dann gar nicht mehr physisch herstellen muss und eine Automation in HA nutzen kann?

**Reply by Trunks1982** [reply]
Genau den sg kontakt brauch es nicht.Du kannst das ja mit einer Automatisierung alles machen.
Hab auch ne PV und erhöhe zb das WW.Alles schön komfortabel

**Reply by Nitrox912** [reply]
Ich könnte mir auch so etwas in der Art vorstellen die tatsächliche Raumtemperatur über HA an die WP zu senden, damit diese nicht nur "fiktiv" die VL-Temperatur errechnet, sondern echte Werte dafür nutzt. Wäre sowas denkbar?

**Reply by Trunks1982** [reply]
Mhm das feht eigentlich nicht.Das einzige was man sich erarbeiten könnte wäre die heizkurve und raumsoll so anzupassen das es ca mit raumtemp passt.Bei mir ist es ca mit 0,3 und raumsoll 22.
Man könnte dann in 0.1 stepnden raumsoll immer anpassen lassen.Kannst ja alles testen und berichten🤣

**Reply by Nitrox912** [reply]
Teile sind bereits bestellt! Ich bin sehr gespannt. :smile:

**Comment by Nitrox912** [comment]
Ich habe gerade alles in Betrieb genommen. Funktionierte auf Anhieb! Sehr geil! Mit meiner Rocon BM2C funktioniert es also genauso.

Frage: Wenn ich die WW-Temperatur einmalig auslösen will, dann setzt er ja kurzzeitig auf 70 Grad, dann aber auf 45 Grad. In der YAML kann ich das ja anpassen, wie aktualisiere ich das dann aber?

**Reply by Trunks1982** [reply]
du muss aber die can nachricht für deine bestimmte ww temp wissen.Wenn du es mit esphome angepasst hast musst du install klicken und dann über wireless auswählen.

Frage wie hat dir meine Anleitung geholfen? 
War sie gut verständlich?

**Reply by Nitrox912** [reply]
Deine Anleitung war sehr gut. Ich würde evtl. noch auf den ESP32 mit bereits verlöteten Kontakten verlinken. Ich habe diesen genommen und entsprechende Jumper-Kabel dazu bestellt und brauchte nur alles zusammenstecken.

Bei der WW-Thematik sehe ich jetzt aber nicht durch. Ich finde 45 Grad etwas wenig. Hier wäre es schön evtl. vorher eine Temperatur einstellen zu können und dann die Bereitung zu starten. Und das ändern der fixen VL-Temperatur wäre auch noch smart. ;-)

**Reply by Trunks1982** [reply]
Ja den ESP32 mit verlöteten Kontakten hab ich auch genommen.Könnte ich reinschreiben. Verlinken weiß ich  nicht so recht ob ich das machen soll.Wäre ja inderekte Werbung.
Ich könnte dir ne extra yaml mit einem höheren Wert schicken.Das müsstest du dann austauschen.
Die WW Taste hatte ich für meine Bedürfnisse bzw für meine Temp erstellt. Ich weiß noch noch nicht wie ich es anders umsetzen könnte.Es würde ja wenn reichen das er die Eingestellte WW Temp nimmt und dahin zurück stellt. Aber gebe dir da recht wäre besser.

Fixe VL Temp Eingabe werde ich noch nachreichen.Welche willst du da haben?
Reicht VL min und VL Max?

**Reply by Nitrox912** [reply]
Ich denke ich mache mir eine eigene Automation in HA, die die Warmwassereinstellung genauso kurz auf 70 und dann meine gewünschte Temperatur einstellt. Ich denke das wäre sinnvoller, um die Wert individuell festlegen zu können.

Fixe VL-Temp macht natürlich nur Sinn, wenn sie auch einstellbar wäre. Ließe sich das auch bewerkstelligen?

Welche Funktion haben die "Heizen Thermostat" und "Kühlen Thermostat" damit kann ich leider nichts anfangen.

Und wie hast du den COP in deinem HA-Beispiel erstellt? Greifst du noch den Stromverbrauch mit z. B. einen Shelly ab?

Die Heizkurve lässt sich nur von 0,2 bis 0,5 verstellen? Am Gerät kann ich sie höher drehen. Hat diese Begrenzung einen Grund?

Ansonsten wirklich tolle Arbeit! Vielen Dank!

**Reply by Trunks1982** [reply]
- Fixe VL-Temp macht natürlich nur Sinn, wenn sie auch einstellbar wäre. Ließe sich das auch bewerkstelligen?
Ja das was man an der Rotex auch einstellen kann.

- Welche Funktion haben die "Heizen Thermostat" und "Kühlen Thermostat" damit kann ich leider nichts anfangen.
Die waren dafür vorgesehen direkt die Schaltkontakte an der Rotex anzusteuern. Da gibt es ja Thermostateingänge für Heizen und Kühlen.

- Und wie hast du den COP in deinem HA-Beispiel erstellt? Greifst du noch den Stromverbrauch mit z. B. einen Shelly ab?
Ja genau mit einem Shelly 3em.Mein verwendetes Script ist aber noch nicht gut. 

- Die Heizkurve lässt sich nur von 0,2 bis 0,5 verstellen? Am Gerät kann ich sie höher drehen. Hat diese Begrenzung einen Grund?
Für mich war der Grund weil ich dachte höher verwendet kein Mensch.Und wo ich angefangen habe musste ich jeden einzelnen Wert händisch eingeben was mit viel Arbeit verbunden war.Kann dies aber mittlerweile erhöhen ohne großen Aufwand.

- Ansonsten wirklich tolle Arbeit! Vielen Dank!
Vielen Herzlichen Dank.

Du darfst gerne im Heizungsforum von deinen Erfahrungen erzählen und Werbung machen:-)

**Reply by Nitrox912** [reply]
Wird bei Gelegenheit gemacht. Übrigens finde ich bei mir nichts um den SG-Kontakt auszulösen. Ist das rausgefallen?

**Reply by Trunks1982** [reply]
Ja weil unflexibel.Kannst mit HA viel flexibler sein.

**Reply by Nitrox912** [reply]
Was mir noch aufgefallen ist, ist das der Wert thermische Leistung beim Warmwasser mir sehr niedrig ist (geringer als die Stromaufnahme der WP) und teilweise sogar Minuswerte angezeigt werden. Wie ist dieser Wert zu verstehen?

**Reply by Trunks1982** [reply]
Minus Werte zeigt er nur an wenn die WP nicht Läuft bzw auch wenn abgetaut wird..Wenn sie läuft sollte alles richtig berechnet werden.Auch da habe ich noch kein optimalen code gefunden.

**Reply by Nitrox912** [reply]
Also beim WW funktioniert der Wert bei mir leider gar nicht. Kommt der Wert "Erzeugte Energie Gesamt" direkt aus der WP oder wird der auch berechnet? 

<img width="1537" alt="Unbenannt" src="https://github.com/Trunks1982/Daikin-Rotex-HPSU-CAN/assets/127048498/36d01f63-14e8-4417-aba0-1298bd42fdf5">

**Reply by Trunks1982** [reply]
Meinst du ww einstellen?
Erzeugte Energie kommt aus der WP ja

**Comment by Trunks1982** [comment]
Achte aufjedenfall immer mal auf neue Updates.Kann sein das ich heute noch wieder eins machen werde.Kannst die dann über die WEBUI einspielen zb

**Reply by Nitrox912** [reply]
Welche Datei lade ich dann immer hoch? Einfach die aktuellste rotex-daikin.bin?

**Reply by Trunks1982** [reply]
Rotex.bin

**Reply by Nitrox912** [reply]
Alles klar... Auch beim Heizen wird die thermische Leistung bei mir nicht richtig berechnet und nur ein viel zu niedriger Wert angezeigt.

**Reply by Trunks1982** [reply]
mhm Also bei mir berechnet er das eigentlich ganz gut.

**Reply by Nitrox912** [reply]
Jetzt läuft die Heizung gerade kräftig und es erscheint ein plausibler Wert. Mit ist aber auch aufgefallen das der Wert "Erzeugte Energie Gesamt" nirgendwo unter Statistik bei mir zu finden ist. Eventuell gibt es hier doch Daikin spezifische Einstellungen zu berücksichtigen. Auch erhalte ich nur sporadisch Werte meiner Umwälzpumpe, obwohl die laut WP kräftig arbeitet und irgendwie hab ich den Eindruck, dass sich die Werte WW und Raum-Soll verstellt haben.

**Reply by Trunks1982** [reply]
Erfragt alle 15sek ab.Und nur dann kommen Werte.Und ja meine Standart einstellungen sind bei Raumsoll 19 grad und ww 45.Werden auch immer wieder so sein wenn der ESP neugestartet wird.

**Reply by Nitrox912** [reply]
Ok, das macht Sinn. Ich habe den ESP aber nicht vom Strom getrennt.
Edit: Das passiert alle paar Minuten mal. Dabei hängt dann auch die rechte Seite im WebUI. Also ob der ESP abstürzt und neu startet und somit die Werte wieder resettet.

**Reply by Trunks1982** [reply]
mhm.Den Fehler kenne ich noch nicht.Frage hast du noch weitere Geräte nach deinem Hauptbedienteil angeschlossen oder Nur das Bedienteil An der Heizung selber und dann den ESP?

**Reply by Nitrox912** [reply]
Ja genau. Ich probiere es noch mal mit einem geschirmten Netzwerkkabel. Ich hatte bisher ein Buskabel genommen. Mal schauen ob sich da was ändert.

**Reply by Trunks1982** [reply]
Ok.

**Reply by Nitrox912** [reply]
Nun passt die Thermische Leistung. Die Umwälzpumpe und Kompressor werden trotz Heizbetrieb nur sporadisch angezeigt.

Die Werte werden aber trotzdem unregelmäßig zurückgesetzt.

**Reply by Trunks1982** [reply]
mhm nur ww und raumsoll? oder auch Heizkurve

**Reply by Nitrox912** [reply]
Alle drei Werte

**Reply by Trunks1982** [reply]
spiel mal bitte die jetzige yaml aus GitHub bitte ein.Kopierst dir alles raus und ersetzt die durch deinen alten yaml code.Bekommst du das hin?

**Reply by Nitrox912** [reply]
Nach mal bitte für Doofe. Ich krieg das sicher hin, verstehe aber nicht ganz was ich wo hin kopieren soll.

**Reply by Trunks1982** [reply]
hast du esphome in HA installiert?

**Reply by Nitrox912** [reply]
ja

**Reply by Trunks1982** [reply]
Gut dann solltest du ja deinen ESP Rotex da sehn.Wenn du da auf edit klickst kannst du ja den code sehen.Den kannst du komplett gegen den von meine githubseite aus der yaml tauschen.Speicherst und sagst dann installieren über wireless

**Reply by Nitrox912** [reply]
Sorry, entweder ich stehe auf dem Schlauch oder da gibt es kein Edit Button.

**Comment by Trunks1982** [comment]
![Bildschirmfoto 2024-04-07 um 00 36 51](https://github.com/Trunks1982/Daikin-Rotex-HPSU-CAN/assets/62701386/aa6193cf-9504-4411-83e7-4c96190d0fd3)

**Reply by Nitrox912** [reply]
Ach das ist ein Add-on! Ich hatte nur die Integration geladen. Ok...

**Reply by Trunks1982** [reply]
ja warte dann schicke ich dir jetzt die neue bin Datei

**Reply by Trunks1982** [reply]
so die bin ist jetzt im Github downloadbar

**Reply by Nitrox912** [reply]
Ok... also ich aktualisiere jetzt nicht die YAML über das HA Addon, sondern über die WebUI vom ESP?

**Reply by Trunks1982** [reply]
genau lädst die bin runter und unter der webbi vom ESP kannst unten bei OTA die bin auswählen und Updaten

**Reply by Nitrox912** [reply]
Ok läuft. Mal gucken, ob sich die Werte wieder zurücksetzen. Für den Kompressorm Thermische Leistung und Umwälzpumpe bekomme ich aber gar keine Werte mehr.

**Reply by Trunks1982** [reply]
ja warte manchmal dauert es.

**Reply by Nitrox912** [reply]
Läuft jetzt stabil! Aber die Erzeugte Energie Gesamt kann bei mir zumindest nicht stimmen. Ich hatte den ESP nochmals zurückgesetzt und neu installiert. Bei mir war danach der Wert wieder auf 28480 kWh zurückgefallen. Ich finden diesen Wert aber auch nicht in meiner Statistik am Gerät.

Und der Status Kompressor ist für mich nachvollziehbar, da er mitunter An anzeigt, obwohl die WP draußen nicht läuft und aus, wenn sie läuft.

**Reply by Trunks1982** [reply]
Dann stimmen bei deiner Anlage die beiden Can Nachrichten nicht.Bei mir und bei den jetzig getesteten anscheinend schon.

**Reply by Nitrox912** [reply]
Lässt sich das irgendwie genauer herausfinden?

**Reply by Trunks1982** [reply]
Nur wenn man das im Netz findet.Ich hab da keine Möglichkeit

**Reply by Trunks1982** [reply]
Wenn du auf i an der Rotex klickst siehst du da im Untermenü nicht die produzierte gesamt Erzeugung?

**Reply by Nitrox912** [reply]
Ich hab eine Daikin Altherma 3 HHT mit Ech20 Speicher. Da sieht das so aus:

![IMG_0174](https://github.com/Trunks1982/Daikin-Rotex-HPSU-CAN/assets/127048498/8aa64deb-8012-49c0-98e2-3e8c40a6a2a4)

<img width="669" alt="Unbenannt3" src="https://github.com/Trunks1982/Daikin-Rotex-HPSU-CAN/assets/127048498/a9e1884d-d85b-42c3-ab8f-e13a35b147fd">

**Reply by Trunks1982** [reply]
Ok hat dann bei dir ne andere Can bus Adresse.

**Reply by rayxxxxx** [reply]
Hi, I have the same Daikin model and I'm looking into this solution to manage it. At the moment, I'm using the ESPAltherma integration (read-only mode). Could you share a wiring/installation scheme of your setup? Do you have any advice on the initial configuration? I was planning to use an M5Stack Atom S3 Lite (ESP32-S3) paired with the Atomic CAN Base. Would that setup work well?

**Reply by wrfz** [reply]
It is always the same wiring schema. You can find it on the main page of this project: https://github.com/Trunks1982/Daikin-Rotex-HPSU-CAN-Seriell

**Comment by Trunks1982** [comment]
Frage welchen Wert setzt du jetzt fix für die vl temp? Kannst mir da auch mal nen screenshot schicken?

**Reply by Nitrox912** [reply]
Aktuell fahre ich 31 °C

Aber wie gesagt, bräuchte ich den Wert ähnlich wie die WW-Temp. verstellbar.

![IMG_0175](https://github.com/Trunks1982/Daikin-Rotex-HPSU-CAN/assets/127048498/7444fddd-2351-42b7-b28a-c8ac5a76f934)
![IMG_0176](https://github.com/Trunks1982/Daikin-Rotex-HPSU-CAN/assets/127048498/364d9d86-da36-4345-8213-95e82fe092f8)

**Reply by Trunks1982** [reply]
Ok danke.

**Reply by Trunks1982** [reply]
Also normalerweise wenn du min vl für witterungsgeführt auf 31 setzt sollte es das gleiche ergeben.

**Reply by Nitrox912** [reply]
Aber wo soll ich denn in deiner Integration die Min.VLT einstellen? Ich finde die Idee die fixe VLT einstellen zu können irgendwie eleganter. Vor allem ist es dann auch eindeutig.

**Reply by Trunks1982** [reply]
Min Vlt und max erstelle ich gerade.Suche gerade die Einstellung für fix Umstellung

**Reply by Trunks1982** [reply]
So neue bin ist da bitte testen.HK Funktion dabei.Bei mir funktioniert es.

**Reply by Nitrox912** [reply]
Ok, mache ich gleich.

Diese Warnung taucht bei mir regelmäßig auf.

<img width="704" alt="4" src="https://github.com/Trunks1982/Daikin-Rotex-HPSU-CAN/assets/127048498/06b228af-ddf6-4261-ad01-044fcf48d872">

**Reply by Trunks1982** [reply]
Nicht schlimm.Kannst ignorieren.

**Reply by Nitrox912** [reply]
Update über die WebUI gemacht und nun geht gar nix mehr. Komme nicht mehr auf den ESP. Update war erfolgreich aber was hab ich falsch gemacht?

**Reply by Trunks1982** [reply]
Hast dir Rotex.bin genommen ne?
Versuch bitte einmal vom Strom nehmen und Neustart.
Wenn das nicht funktioniert bitte komplett neu flashen.

**Reply by Nitrox912** [reply]
Läuft wieder und die Verstellung funktioniert auch! :)

**Reply by Nitrox912** [reply]
Alllerdings werden bei thermische Leistung, Umwälzpumpe und Kopressor Status keine Werte mehr angezeigt.

**Reply by Trunks1982** [reply]
Wie gesagt bei mir funktioniert das alles.Muss dann wenn mit deiner BM2 zusammenhängen.

**Reply by Nitrox912** [reply]
Die thermische Leistung errechne ich mir jetzt über ein HA Template und das sieht ganz gut aus. Alle Werte wie VL, RL und Durchfluss funktionieren ja 1A. Hab ich eigentlich die Möglichkeit Werte, die mit meiner System nicht sauber funktionieren aus der Abfrage vom ESP rauszuwerfen?

**Reply by Trunks1982** [reply]
Ja da musst du in die yaml eingreifen und wenn du  # vor jede Zeile packst kommentierst du die aus.

**Reply by Nitrox912** [reply]
Und wie kann ich daraus eine .bin erstellen? Über das HA Addon die YAML aktualisieren hat bisher nicht geklappt.

**Reply by Trunks1982** [reply]
Nur über das addon.Da kannst die yaml bearbeiten und draufspielen.Nur bedenke bei jedem update musst entweder teile der yaml tauschen oder nach bin einspielen wieder auskommentieren.

**Comment by Nitrox912** [comment]
Wenn ich mir die YAML genauer anschaue, dann denke ich, dass die "Thermische Leistung" nicht richtig angezeigt wird, weil die an den Status vom Kompressor gekoppelt ist. Kannst du mir sagen, warum die in deiner Formel mit 4,91 * Durchflussrate rechnest? In meiner Formel habe ich eine spezifische Wärmekapazität von 4,186 Joule Energie, mit der ich rechne.

**Reply by Trunks1982** [reply]
Hatte ich irgendwo so gefunden wie es berechnet wird.

**Reply by Trunks1982** [reply]
Kann aber auch sein das da nen Fhler drin ist zahlen dreher

**Comment by Nitrox912** [comment]
Wie oft sendet der ESP die Befehle zur WP? Ich habe ständig Statusmeldung im Log, obwohl ich nichts verändert habe und die WP nicht läuft:

<img width="271" alt="5" src="https://github.com/Trunks1982/Daikin-Rotex-HPSU-CAN/assets/127048498/d817a7ef-67b2-4acd-8c28-fa6fe799fc63">

**Reply by Trunks1982** [reply]
Alle 15sek

**Reply by Nitrox912** [reply]
Warum wird das so gemacht? Reicht es nicht den Befehl nur bei Eingabe einer Änderung zu senden?

**Reply by Trunks1982** [reply]
Das tut er ja.Es werden nur werte abgefragt nichts verstellt.Wird nur nach eingabe was verstellt.

**Reply by Nitrox912** [reply]
Ah ok, dann habe ich das missverstanden.

**Comment by Nitrox912** [comment]
Wie ist dein Plan für die weitere Entwicklung? Welche Funktionen willst du alle noch implementieren?

**Reply by Trunks1982** [reply]
Eigentlich nur noch das was sinn macht.Klar ich könnte das komplette Menü abfragen aber macht das sinn ich glaube nicht.Und das strapaziert ja auch den Canbus.

**Reply by Nitrox912** [reply]
Und an welchen arbeitest du gerade noch?

**Reply by Nitrox912** [reply]
Weißt du schon wann du die Einstellung von im min./max. und fix Vorlauf implementieren kannst? Kann man dich irgendwie unterstützen?

**Reply by Trunks1982** [reply]
Nein weiss ich noch nicht.Wenn man sich mit dem code auskennt ja aber sonst leider nicht.

**Comment by Roadblock1987** [comment]
Hallo zusammen, 

bin grad dabei meine WP auch "smart" zu machen. Idee war schon länger da, aber Umsetzung bisher nicht angegangen. Hab dabei das Projekt  hier gefunden und soweit alles zusammengetragen und zusammengebaut. Dazu zunächste erstmal danke für die Mühe ;)

Anzeigen der Rotex WEBUI und einbinden in HA hat soweit auch alles geklappt. Leider bekomme ich keine Werte. Weder in WEBUI noch in HA. bekomme nur angefügte Statusmeldungen. Bin dann beim Suchen auf diese Anfrage gestoßen, da ich die selbe WP mit BM2C verwende.

Wo könnte mein Fehler liegen? Die WP braucht nicht zusätzlich ne Info, dass J13 verwendet werden soll, der ist ja aktuell leer und für externe Bedienung vorgesehen?

Viele Grüße 

![Bild_2024-10-06_095022273](https://github.com/user-attachments/assets/fa81fe21-55f6-41a0-a601-52fccc0d26ef)

**Reply by Trunks1982** [reply]
Hi vielleicht Kabel falsch angeschlossen?

**Reply by wrfz** [reply]
Hi @Roadblock1987,

die grüne Zeile `sendGet` zeigt, dass der ESP eine Anfrage an die Pumpe sendet, um den `1 x Warmwasser` Wert zu erhalten.
Würde alles richtig funktionieren, würdest du von der Pumpe an den ESP eine `handle` - Antwort sehen, wie bei diesem Beispiel:
```
10:27:37 [I] [sendGet:178] 1 x Warmwasser can_id<0x680> command<31 00 FA 01 44 00 00>
10:27:37 [I] [handle :178] 1 x Warmwasser<Aus> can_id<0x180> data<D2 00 FA 01 44 00 00>
```
Da die Antwort von der Pumpe nicht zurück kommt, gibt es dafür folgende mögliche Ursachen:
- RX/TX Konfiguration der GPIO-Pins in deiner yaml, passt nicht zu den Pins, die du auf dem ESP-Board verwendest
- Etwas beim Verdrahten passt nicht (z.B. RX/TX zwischen CAN-Tranciever und der Pumpe oder dem ESP)
- Verwendete GPIOs auf dem ESP sind defekt oder für den Einsatzzweck nicht vorgesehen (könntest andere mal ausprobieren)
- Serielle Schnittstelle der Pumpe defekt (wollen wir davon mal nicht ausgehen)

**Comment by Roadblock1987** [comment]
Hi,

danke für die schnelle Rückmeldung. Ich vermute da irgendwas mit meinen GPIOs.

Ich habe die Rotex factory mit tx05 tx06 im Namen installiert. Also die letzte im BIN Ordner. Verdrahtet habe ich nach Anleitung GPIO 47 und 48. Gucke ich im HA erscheint Version 2024.9.1 und dort in den examples sind die GPIOs 2 und 3 unter CAN-Bus eingetragen.

Suche ich hier an falscher Stelle oder ergibt das irgendeinen Sinn? 

Versuche ich die Version 2024.9.6 zu installieren, startet der ESP nicht vernünftig bzw. komme ich nicht in das WLAN.

Hoffe ich komme der Lösung näher? Wo könnte ich die entsprechenden GPIos einstellen, damit der ESP das übernimmt....

Grüße

**Reply by Trunks1982** [reply]
Oh schliess bitte so an wie im Namen gekennzeichnet. Sag bescheit wenn das klappt, dann miss ich die doku sofort anpassen.

**Reply by Roadblock1987** [reply]
Hi, Anschluss über GPIO 5 und 6 (CTX/CRX) war erfolgreich. Werte wurden direkt erkannt.

Vielen Dank!

Zusätzlich kurze Frage, wie versorgt ihr die ESP in den Pumpen? Steckernetzteil? Zusätzliches kleines Netzteil an einer der Phasen? Ich gehe davon aus die Versorgungsspannung vom CAN (VCC und GND, Pin 3 und 4)wird nicht reichen?

Grüße

**Reply by Trunks1982** [reply]
Freut mich das es funktioniert. Ich persönlich habe den esp 3m weiter weg von Rotex und betreibe in ganz normal über ein USB Netzteil.

**Reply by wrfz** [reply]
Da du HA betreibst, bietet es sich an ESPHome dort zu installieren.
Damit kannst du die HPSU Anwendung für deinen ESP konfigurieren und deinen Bedürfnissen anpassen.

**Reply by Roadblock1987** [reply]
Ich guck mir das die Tage nochmal an was alles möglich ist. Wichtig ist nur erstmal das alles soweit läuft. Danke euch für die Mühe

**Comment by Sguerra88** [comment]
Hi, first of all congratulations on the work I'm trying an integration with Dakin altherma 3 R ech20 (EHSX08p50DA) I managed to do everything very quickly I opted to buy the M5 modules, the connections are all correct but I also find this error, and without the values ​​read, what can I do? The device is mounted near the Rocon U1 about 10 meters from the machine, there is also a Rocon G1 on the can bus. Could they cause problems?
![image](https://github.com/user-attachments/assets/e45e51e4-aeda-4513-a736-6870b7837156)
thanks 
Stefano
